src/Components/ProductList/ProductList.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import axios from 'axios';
+import OfferBox from 'components/OfferBox/OfferBox';
+
+const baseURL = 'https://61a7c977387ab200171d2ec9.mockapi.io/list';
+
+class ProductList extends React.Component {
```

Comment:
zrobiłabym całość na hookach https://pl.reactjs.org/docs/hooks-reference.html - czyli tak jak te poprzednie :) Klasa tutaj nie jest potrzebna, a pozbędziesz się `this`